clang/test/SemaCXX/warn-unsafe-buffer-usage-fixits-parm-main.cpp

```diff
@@ -8,6 +8,5 @@
 // main function
 int main(int argc, char *argv[]) { // expected-warning{{'argv' is an unsafe pointer used for buffer access}}
   char tmp;
-  tmp = argv[5][5];                // expected-note{{used in buffer access here}} \
-				      expected-warning{{unsafe buffer access}}
```

Comment:
We need this warning. `argv` is a two-dimensional pointer and we don't know the bounds.

Reply:
Ah! Yes you are right.

Reply:
The warning is now attached to the declaration and clang now emits a note here. This is because we are now associating both (on both dimensions) array accesses to the DRE. I thought that will make the programming model more consistent. WDYT?

Reply:
I see, makes sense. I didn't realize that's what's happening and only saw warnings going missing in tests.
I would prefer changing the diagnostics structure in a separate patch and only suppressing the false positives for now.
Do you think we could do it that way?

Reply:
Yes. That is doable. I can just take out the part that associates the operations to DREs for 2-D array accesses. Perhaps we should file a radar to claim operations that are clearly associated with DREs?

Reply:
Resolved in the latest diff.